crypto/bio/bss_dgram.c

```diff
@@ -295,17 +295,19 @@ static void dgram_adjust_rcv_timeout(BIO *b)
         int sz = sizeof(timeout);
         if (getsockopt(b->num, SOL_SOCKET, SO_RCVTIMEO,
                        (void *)&timeout, &sz) < 0) {
-            perror("getsockopt");
+            ERR_raise_data(ERR_LIB_SYS, get_last_socket_error(),
+                           "calling getsockopt()");
         } else {
             data->socket_timeout.tv_sec = timeout / 1000;
             data->socket_timeout.tv_usec = (timeout % 1000) * 1000;
         }
 #  else
         socklen_t sz = sizeof(data->socket_timeout);
         if (getsockopt(b->num, SOL_SOCKET, SO_RCVTIMEO,
-                       &(data->socket_timeout), &sz) < 0) {
-            perror("getsockopt");
-        } else
+                       &(data->socket_timeout), &sz) < 0)
+            ERR_raise_data(ERR_LIB_SYS, get_last_socket_error(),
+                           "calling getsockopt()");
+        else
             OPENSSL_assert(sz <= sizeof(data->socket_timeout));
```

Comment:
Side note: We really should not use OPENSSL_assert() anywhere inside the library. Failing the assert causes a crash even in a non-debug build. Not for this PR though.

Reply:
We really should clean these up.

```sh
$ egrep -r -w '(OPENSSL_)?assert' crypto providers ssl include apps | wc -l
401
```